lib/logstash/inputs/redis.rb

```diff
@@ -217,6 +246,86 @@ def list_runner(output_queue)
     end
   end
 
+  # private
+  def list_pattern_stop
+    return if @redis.nil? || !@redis.connected?
+
+    @redis.quit rescue nil
+    @redis = nil
+    @threadpool.shutdown
+    @threadpool.wait_for_termination
+  end
+
+  # private
+  def worker_consume(output_queue, key)
+    @logger.debug("Launched worker for #{key}")
+    redis = new_redis_instance
+    begin
+      (0..@pattern_list_max_items-1).each do
+        if stop?
+          @logger.debug("Breaking from thread #{key} as it was requested to stop")
+          break
+        end
+        value = redis.lpop(key)
+        break if value.nil?
+        queue_event(value, output_queue)
+      end
+    ensure
+      redis.quit rescue nil
```

Comment:
`... rescue nil` is a bit evil as it catches any error, including making a typo in the variable on the left side (`rediss.quit`) which would hide broken things not so conveniently. Probably not a huge deal in the two use cases I saw here, but it's something that linters prevent nowadays.